src/android/io/branch/BranchSDK.java

```diff
@@ -54,9 +54,10 @@ public void onNewIntent(Intent intent)
         this.activity = this.cordova.getActivity();
         this.activity.setIntent(intent);
 
-        if (this.activity != null) {
-            this.initSession(null);
-        }
+        // HURDLR CHANGE (PV): we don't need to initSession because we'll do it ourselves in javascript
```

Comment:
We call initSession() on AppResume (once the JS is loaded up). That way, we're able to preserve what the latest parameters was as our promises get resolved and we potentially store information in our cache if we don't need it yet.